detectable/src/main/java/com/synopsys/integration/detectable/detectables/bitbake/parse/BitbakeGraphTransformer.java

```diff
@@ -73,4 +89,42 @@ public DependencyGraph transform(final BitbakeGraph bitbakeGraph) {
         return dependencyGraph;
     }
 
+    private Optional<ExternalId> generateExternalId(final String name, final String version, final Map<String, BitbakeRecipe> componentLayerMap, final Map<String, Integer> layerPriorityMap) {
+        final BitbakeRecipe bitbakeRecipe = componentLayerMap.get(name);
+        ExternalId externalId = null;
+
+        if (bitbakeRecipe == null) {
+            if (componentLayerMap.containsKey(name)) {
+                logger.warn(String.format("Component '%s==%s' is in the component layer map. But potential layers were not populated. This should be reported.", name, version));
+            } else {
+                logger.debug(String.format("Failed to find component '%s' in component layer map.", name));
+            }
+            if (name.endsWith(NATIVE_SUFFIX)) {
+                final String alternativeName = name.replace(NATIVE_SUFFIX, "");
+                logger.debug(String.format("Generating alternative component name '%s' for '%s==%s'", alternativeName, name, version));
+                externalId = generateExternalId(alternativeName, version, componentLayerMap, layerPriorityMap).orElse(null);
+            } else {
+                logger.debug(String.format("'%s==%s' is not an actual component. Excluding from graph.", name, version));
+            }
+        } else {
+            final Optional<BitbakeRecipe.Layer> highestPriorityLayer = getHighestPriorityLayer(bitbakeRecipe, layerPriorityMap);
```

Comment:
Can't you calculate highest priority layer once? It doesn't seem specific to the external id.

Reply:
It is specific to component name. For each of which an external id is generated. So one could argue it is specific to external id.